src/elexclarity/formatters/results.py

```diff
@@ -73,19 +73,46 @@ def aggregate_choice_vote_types(self, choice, level):
 
         return subunits
 
-    def format_subunits(self, choices, level):
+    def get_vote_totals_by_vote_types(self, choices, level):
+        """
+        Creates a mapping from subunit IDs to the total number of votes in
+        for that vote type across all choices.
+        """
+        subunit_vote_types = {}
+        clarity_level = level.capitalize()
+
+        for choice in choices:
+            for vote_type in get_list(choice.get("VoteType", [])):
+                vote_type_subunits = vote_type.get(clarity_level, [])
+                for vote_type_subunit in get_list(vote_type_subunits):
+                    subunit_id = self.get_precinct_id(vote_type_subunit["name"])
+                    subunit_vote_types.setdefault(subunit_id, defaultdict(lambda: 0))
+                    subunit_vote_types[subunit_id][vote_type["name"]] += int(vote_type_subunit["votes"])
+
+        return subunit_vote_types
+
+    def _get_valid_contest_choices(self, contest):
+        return list(filter(lambda choice: choice.get("text"), get_list(contest["Choice"])))
+
+    def format_subunits(self, choices, level, subunit_fully_reporting_statuses=None):
         """
         Takes a list of `Choice` objects from Clarity and aggregates/transforms
         them into a the format our data importer expects.
         """
         subunit_results = {}
-        for choice in filter(lambda choice: choice.get("text"), choices):
+
+        for choice in choices:
             choice_votes_by_subunit = self.aggregate_choice_vote_types(choice, level)
             for subunit_id, subunit_choice_votes in choice_votes_by_subunit.items():
                 subunit_results.setdefault(subunit_id, {"id": subunit_id, "counts": defaultdict(lambda: 0)})
                 choice_id = self.get_choice_id(choice.get("text"))
                 subunit_results[subunit_id]["counts"][choice_id] += subunit_choice_votes
 
+        if subunit_fully_reporting_statuses:
+            for subunit_id, subunit_result in subunit_results.items():
+                is_subunit_fully_reporting = subunit_fully_reporting_statuses.get(subunit_id, False)
+                subunit_result["precinctsReportingPct"] = 100 if is_subunit_fully_reporting else 0
+                subunit_result["expectedVotes"] = sum(subunit_results[subunit_id]["counts"].values()) if is_subunit_fully_reporting else 0
```

Comment:
Out of curiosity, why set `expectedVotes` here? Even if a subunit hasn't reported, it feels like we would still expect a non-0 number of votes at some point in the future? I would suggest using `None` instead of `0` (or not setting it at all) if the precinct has not yet reported, since we already have existing logic in the render engine to handle those situations.

(Also, I think this might risk some divide-by-zero errors when calculating expected vote percents? I can't remember if we check for that in every situation.)

Reply:
I agree, None (or not setting it) would be better, though we aren't actually going to look at this data in the render engine anytime soon and I thought we had been using 0 for expectedVotes in other places 🤷 In any case, I will update!